x-pack/plugins/xpack_main/server/lib/telemetry/monitoring/get_beats_stats.js

```diff
@@ -105,11 +105,21 @@ export function processResults(results = [], { clusters, clusterHostSets, cluste
 
       const stateHost = get(hit, '_source.beats_state.state.host');
       if (stateHost !== undefined) {
-        const hostSet = clusterArchitectureSets[clusterUuid];
-        const hostArchPlatform = `${stateHost.architecture}/${stateHost.os.platform}`;
-        hostSet.add(hostArchPlatform);
-        clusters[clusterUuid].architecture.names = Array.from(hostSet);
+        const hostMap = clusterArchitectureMaps[clusterUuid];
+        const hostKey = `${stateHost.architecture}/${stateHost.os.platform}`;
+        let os = hostMap.get(hostKey);
+
+        if (!os) { // undefined if new
+          os = { name: stateHost.os.platform, architecture: stateHost.architecture, count: 0 };
+          hostMap.set(hostKey, os);
+        }
+
+        // total per os/arch
+        os.count += 1;
+
+        // overall total (which should be the same number as the sum of all os.count values)
         clusters[clusterUuid].architecture.count += 1;
+        clusters[clusterUuid].architecture.architectures = Array.from(hostMap.keys());
```

Comment:
My intent was for the `architectures` to be equal to the _values_ from the map, not the _keys_. (It would be more efficient to do what you were doing for just the keys).

```js
clusters[clusterUuid].architecture.architectures = Array.from(hostMap.values());
```

Reply:
Ahhhhh. I finally understand!